src/trio/_tests/test_util.py

```diff
@@ -323,25 +324,36 @@ async def test_raise_single_exception_from_group() -> None:
         [
             ValueError("foo"),
             ValueError("bar"),
-            KeyboardInterrupt("this exc doesn't get reraised"),
+            KeyboardInterrupt("is not reraised but preserve error msg"),
         ],
     )
-    with pytest.raises(KeyboardInterrupt, match=r"^$") as excinfo:
+    with pytest.raises(
+        KeyboardInterrupt,
+        match=r"is not reraised but preserve error msg",
+    ) as excinfo:
         raise_single_exception_from_group(eg_ki)
+
     assert excinfo.value.__cause__ is eg_ki
     assert excinfo.value.__context__ is None
 
-    # and same for SystemExit
+    # and same for SystemExit but verify code too
+    sysexit_exc = SystemExit("preserve error code")
+    sysexit_exc.code = 1  # can't set it in constructor
```

Comment:
Use a code that isn't `1` because that's what it defaults to iirc

(Also I think code is the 1st arg of `SystemExit`?)

Reply:
I agree with what this is testing, I think the comment should be changed though.